### ISSUE TITLE
Mouse support

### ISSUE THREAD

**Issue by adelarsq** [ap/vim-buftabline#6]
When doing a double click to change between buffers the tabline looks like:
![image](https://cloud.githubusercontent.com/assets/430272/6001852/d6e29fbc-aacc-11e4-814a-052bef4042d2.png)
Would be nice support mouse click.

**Comment by ap** [comment]
Yes, it would be nice. But it’s very difficult to make it work without glitches. And turning on this feature would make it *impossible* to use regular Vim tabs at all. If you’re interested, you can try picking up the [`krazy` branch](/ap/vim-buftabline/tree/krazy) and finishing what I started, and I shall be happy to look at the results. But I do not intend to put further work into it myself, sorry.

**Comment by adelarsq** [comment]
It's more difficult than I thought https://github.com/bling/vim-airline/issues/369
Anyway, thanks for the quick response.

**Comment by ap** [comment]
I don’t know that it’s impossible; it could be that sufficiently clever hackery can make it work, as long as you are willing to give up Vim tabs completely forever.

Then the basic approach is to have as many extra Vim tabs as you have buffers. You can then intercept the tab change events caused by the user clicking on the tabline, and switch buffers accordingly.

The problem is making sure that the buffer and window management works exactly like it would if you only had a single Vim tab. E.g. keeping Vim from creating or not-closing empty buffers in various circumstances.

I got most of the way there with several different approaches, each with another combination of tricks to create the right illusion, but each approach had its own cases in which the illusion broke. I didn’t manage to make any of them work perfectly. In the end I gave up.

Someday in the future I may try again, and one of these attempts may even succeed.
